### PR TITLE
[msal-common] Enable "-strict" in tsconfig for msal-common

### PR DESCRIPTION
**Pull request by sameerag**
Enabling the below flags in `.tsconfig` for `msal-common`:

```javascript

        "strictNullChecks": true,
        "strictFunctionTypes": true,
        "strictPropertyInitialization": false,
        "strict": true,
        "alwaysStrict": true,
        "noImplicitThis": true,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "noFallthroughCasesInSwitch": true,
```

### CODE REVIEW

**Comment by sameerag** [comment]
@tnorling We may want to retest `ClientCapabilities` after this PR. If E2E exists, great, else let us know, will alert you.

File: lib/msal-common/src/client/RefreshTokenClient.ts [hunk 1]
```diff
@@ -154,26 +154,34 @@ export class RefreshTokenClient extends BaseClient {
 
         parameterBuilder.addClientInfo();
 
-        const correlationId = request.correlationId || this.config.cryptoInterface.createNewGuid();
+        const correlationId = request.correlationId ? request.correlationId : this.config.cryptoInterface.createNewGuid();
```
Comment (jasonnutter) [inline lib/msal-common/src/client/RefreshTokenClient.ts:157]: These are functionally equivalent, what was the warning/error?

Comment (sameerag) [reply lib/msal-common/src/client/RefreshTokenClient.ts:157]: request.correlationId can be `undefined`

**Comment by sameerag** [comment]
This PR is abandoned, we have a new PR addressing this with some design changes made in the `request` APIs.

**Pull request closed without merging**